### PR TITLE
media-gfx/freecad: version bump freecad-0.16

### PR DESCRIPTION
**Pull request by bdouxx**
- change SRC_URI to github
 - remove bindist : https://bugs.gentoo.org/show_bug.cgi?id=580256
 - 3rdParty :  boost and pivy are not in the src, so don't need to remove them

Package-Manager: portage-2.3.3

### CODE REVIEW

File: media-gfx/freecad/freecad-0.16.ebuild [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+HOMEPAGE="http://www.freecadweb.org/"
+SRC_URI="https://github.com/FreeCAD/FreeCAD/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+COMMON_DEPEND="dev-cpp/eigen:3
+	dev-libs/boost
```
Comment (SoapGentoo) [inline media-gfx/freecad/freecad-0.16.ebuild]: needs `:=`

File: media-gfx/freecad/freecad-0.16.ebuild [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+S="${WORKDIR}/FreeCAD-${PV}"
+
+DOCS=(README.md ChangeLog.txt)
+
+pkg_setup() {
+	fortran-2_pkg_setup
+	python-single-r1_pkg_setup
+
+	[ -z "${CASROOT}" ] && die "empty \$CASROOT, run eselect opencascade set or define otherwise"
```
Comment (SoapGentoo) [inline media-gfx/freecad/freecad-0.16.ebuild]: always use bash `[[ ]]` and not POSIX `[ ]`

File: media-gfx/freecad/freecad-0.16.ebuild [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+}
+
+src_prepare() {
+	epatch "${FILESDIR}"/${PN}-0.14.3702-install-paths.patch
```
Comment (SoapGentoo) [inline media-gfx/freecad/freecad-0.16.ebuild]: eschew `epatch` in EAPI 6, stick it in `PATCHES`, call 1. `default` 2. the `sed` line and remove `eapply_user`

File: media-gfx/freecad/freecad-0.16.ebuild [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+src_configure() {
+	export QT_SELECT=4
+	local mycmakeargs=(
+		-DOCC_INCLUDE_DIR="${CASROOT}"/inc
+		-DOCC_INCLUDE_PATH="${CASROOT}"/inc
+		-DOCC_LIBRARY="${CASROOT}"/lib/libTKernel.so
+		-DOCC_LIBRARY_DIR="${CASROOT}"/lib
+		-DOCC_LIB_PATH="${CASROOT}"/lib
+		-DCOIN3D_INCLUDE_DIR="${EROOT}"usr/include/coin
```
Comment (SoapGentoo) [inline media-gfx/freecad/freecad-0.16.ebuild]: PMS violation - `EROOT` cannot be used in `src_*` functions (and also makes no sense in `src_*` functions). Replace it with `EPREFIX`, and don't forget the slash!

File: media-gfx/freecad/freecad-0.16.ebuild [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+		-DCOIN3D_LIBRARY="${EROOT}"usr/$(get_libdir)/libCoin.so
+		-DSOQT_LIBRARY="${EROOT}"usr/$(get_libdir)/libSoQt.so
+		-DSOQT_INCLUDE_PATH="${EROOT}"usr/include/coin
+		-DCMAKE_INSTALL_PREFIX="${EROOT}"usr/$(get_libdir)/${P}
+		-DCMAKE_INSTALL_DATADIR="${EROOT}"usr/share/${P}/
```
Comment (SoapGentoo) [inline media-gfx/freecad/freecad-0.16.ebuild]: `CMAKE_INSTALL_*DIR` type commands should be used relative (i.e. `share/${P}`) and not in an absolute way

File: media-gfx/freecad/freecad-0.16.ebuild [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+		-DOCC_INCLUDE_DIR="${CASROOT}"/inc
+		-DOCC_INCLUDE_PATH="${CASROOT}"/inc
+		-DOCC_LIBRARY="${CASROOT}"/lib/libTKernel.so
+		-DOCC_LIBRARY_DIR="${CASROOT}"/lib
+		-DOCC_LIB_PATH="${CASROOT}"/lib
+		-DCOIN3D_INCLUDE_DIR="${EROOT}"usr/include/coin
+		-DCOIN3D_LIBRARY="${EROOT}"usr/$(get_libdir)/libCoin.so
+		-DSOQT_LIBRARY="${EROOT}"usr/$(get_libdir)/libSoQt.so
+		-DSOQT_INCLUDE_PATH="${EROOT}"usr/include/coin
+		-DCMAKE_INSTALL_PREFIX="${EROOT}"usr/$(get_libdir)/${P}
```
Comment (SoapGentoo) [inline media-gfx/freecad/freecad-0.16.ebuild]: defined by cmake-utils, don't need to define

File: media-gfx/freecad/freecad-0.16.ebuild [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+	#                (external salomesmesh is available in "science" overlay)
+
+	cmake-utils_src_configure
+	ewarn "${P} will be built against opencascade version ${CASROOT}"
+}
+
+src_install() {
+	cmake-utils_src_install
+
+	prune_libtool_files
```
Comment (SoapGentoo) [inline media-gfx/freecad/freecad-0.16.ebuild]: you sure this is needed? a CMake-based build system invoking libtool?

File: media-gfx/freecad/freecad-0.16.ebuild [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+	make_wrapper FreeCAD \
+		"${EROOT}"usr/$(get_libdir)/${P}/bin/FreeCAD \
```
Comment (SoapGentoo) [inline media-gfx/freecad/freecad-0.16.ebuild]: again, `EROOT` is wrong here

File: media-gfx/freecad/freecad-0.16.ebuild [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+		"" "${EROOT}"usr/$(get_libdir)/${P}/lib
+
+	make_desktop_entry FreeCAD "FreeCAD" "" "" "MimeType=application/x-extension-fcstd;"
+
+	# install mimetype for FreeCAD files
+	insinto /usr/share/mime/packages
+	newins "${FILESDIR}"/${PN}.sharedmimeinfo "${PN}.xml"
+
+	# install icons to correct place rather than /usr/share/freecad
+	pushd "${ED}/usr/share/${P}"
```
Comment (SoapGentoo) [inline media-gfx/freecad/freecad-0.16.ebuild]: missing `|| die`, also ED is slash terminated, please use `${ED%/}/`

File: media-gfx/freecad/freecad-0.16.ebuild [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+	for size in 16 32 48 64; do
```
Comment (SoapGentoo) [inline media-gfx/freecad/freecad-0.16.ebuild]: make size `local` before

File: media-gfx/freecad/freecad-0.16.ebuild [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+		newicon -s ${size} freecad-icon-${size}.png freecad.png
+	done
+	doicon -s scalable freecad.svg
+	newicon -s 64 -c mimetypes freecad-doc.png application-x-extension-fcstd.png
+	popd
```
Comment (SoapGentoo) [inline media-gfx/freecad/freecad-0.16.ebuild]: missing `|| die`

File: media-gfx/freecad/freecad-0.16.ebuild [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+
+	# disable compression of QT assistant help files
+	>> "${ED}"usr/share/doc/${P}/freecad.qhc.ecompress.skip
```
Comment (SoapGentoo) [inline media-gfx/freecad/freecad-0.16.ebuild]: use touch instead, also missing `|| die`

File: media-gfx/freecad/freecad-0.16.ebuild [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+	>> "${ED}"usr/share/doc/${P}/freecad.qch.ecompress.skip
+
+	python_optimize "${ED}"usr/{$(get_libdir),share}/${P}/Mod/
```
Comment (SoapGentoo) [inline media-gfx/freecad/freecad-0.16.ebuild]: Please use `"${ED%/}"/` style everywhere, as it is more forward compatible (last section, https://devmanual.gentoo.org/ebuild-writing/variables/)

File: media-gfx/freecad/freecad-0.16.ebuild [hunk 14]
```diff
@@ -0,0 +1,143 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 )
+
+inherit cmake-utils eutils fortran-2 multilib python-single-r1 fdo-mime
```
Comment (SoapGentoo) [inline media-gfx/freecad/freecad-0.16.ebuild]: I don't see the need for `eutils.eclass` after purging `epatch` and `prune_libtool_files` and `multilib.eclass` anymore

Comment (bdouxx) [inline media-gfx/freecad/freecad-0.16.ebuild]: done

File: media-gfx/freecad/freecad-0.16.ebuild (hunk 1 above)
Comment (bdouxx) [inline media-gfx/freecad/freecad-0.16.ebuild]: sorry, but I don't really understand what to do
It 's "dev-libs/boost:="  or "dev-cpp/eigen:3=" ?

File: media-gfx/freecad/freecad-0.16.ebuild (hunk 2 above)
Comment (bdouxx) [inline media-gfx/freecad/freecad-0.16.ebuild]: done

File: media-gfx/freecad/freecad-0.16.ebuild (hunk 3 above)
Comment (bdouxx) [inline media-gfx/freecad/freecad-0.16.ebuild]: done

File: media-gfx/freecad/freecad-0.16.ebuild (hunk 4 above)
Comment (bdouxx) [inline media-gfx/freecad/freecad-0.16.ebuild]: done

File: media-gfx/freecad/freecad-0.16.ebuild (hunk 6 above)
Comment (bdouxx) [inline media-gfx/freecad/freecad-0.16.ebuild]: I need to remove line with CMAKE_INSTALL_PREFIX?

File: media-gfx/freecad/freecad-0.16.ebuild (hunk 5 above)
Comment (bdouxx) [inline media-gfx/freecad/freecad-0.16.ebuild]: done

File: media-gfx/freecad/freecad-0.16.ebuild (hunk 7 above)
Comment (bdouxx) [inline media-gfx/freecad/freecad-0.16.ebuild]: delete

File: media-gfx/freecad/freecad-0.16.ebuild (hunk 8 above)
Comment (bdouxx) [inline media-gfx/freecad/freecad-0.16.ebuild]: done

File: media-gfx/freecad/freecad-0.16.ebuild (hunk 9 above)
Comment (bdouxx) [inline media-gfx/freecad/freecad-0.16.ebuild]: corrected

File: media-gfx/freecad/freecad-0.16.ebuild (hunk 11 above)
Comment (bdouxx) [inline media-gfx/freecad/freecad-0.16.ebuild]: done

File: media-gfx/freecad/freecad-0.16.ebuild (hunk 12 above)
Comment (bdouxx) [inline media-gfx/freecad/freecad-0.16.ebuild]: removed

File: media-gfx/freecad/freecad-0.16.ebuild (hunk 13 above)
Comment (bdouxx) [inline media-gfx/freecad/freecad-0.16.ebuild]: done

File: media-gfx/freecad/freecad-0.16.ebuild (hunk 1 above)
Comment (SoapGentoo) [inline media-gfx/freecad/freecad-0.16.ebuild]: `dev-libs/boost:=`, eigen does not need slot operators

File: media-gfx/freecad/freecad-0.16.ebuild (hunk 6 above)
Comment (SoapGentoo) [inline media-gfx/freecad/freecad-0.16.ebuild]: yes, chuck it

File: media-gfx/freecad/freecad-0.16.ebuild [hunk 15]
```diff
@@ -0,0 +1,135 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 )
+
+inherit cmake-utils fortran-2 python-single-r1 fdo-mime
```
Comment (SoapGentoo) [inline media-gfx/freecad/freecad-0.16.ebuild]: sort these alphabetically

File: media-gfx/freecad/freecad-0.16.ebuild [hunk 16]
```diff
@@ -0,0 +1,135 @@
...
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+#sci-libs/orocos_kdl waiting for Bug 604130 (keyword ~x86)
+#dev-qt/qtgui:4[-egl] and dev-qt/qtopengl:4[-egl] : Bug 564978 
+COMMON_DEPEND="dev-cpp/eigen:3
+	dev-libs/boost[python]
```
Comment (SoapGentoo) [inline media-gfx/freecad/freecad-0.16.ebuild]: make this `dev-libs/boost:=[python,${PYTHON_USEDEP}]` in order to enforce matching python impls

File: media-gfx/freecad/freecad-0.16.ebuild [hunk 17]
```diff
@@ -0,0 +1,135 @@
...
+	dev-java/xerces
+	${PYTHON_DEPS}"
+RDEPEND="${COMMON_DEPEND}
+	dev-qt/assistant:4
+	dev-python/pivy
+	dev-python/numpy"
+DEPEND="${COMMON_DEPEND}
+	dev-python/pyside-tools
+	>=dev-lang/swig-2.0.4-r1:0"
+
```
Comment (SoapGentoo) [inline media-gfx/freecad/freecad-0.16.ebuild]: you're missing `REQUIRED_USE="${PYTHON_REQUIRED_USE}"` which is required for `python*-r1.eclass`

File: media-gfx/freecad/freecad-0.16.ebuild [hunk 18]
```diff
@@ -0,0 +1,135 @@
...
+#-DOCC_* defined with cMake/FindOpenCasCade.cmake
+#-DCOIN3D_* defined with cMake/FindCoin3D.cmake
+#-DSOQT_ not used
+#-DFREECAD_USE_EXTERNAL_KDL="ON" waiting for Bug 604130 (keyword ~x86)
+	local mycmakeargs=(
+		-DOCC_INCLUDE_DIR="${CASROOT}"/inc
+		-DOCC_LIBRARY_DIR="${CASROOT}"/$(get_libdir)
+		-DCMAKE_INSTALL_PREFIX="${EPREFIX}"/usr/$(get_libdir)/${P}
+		-DCMAKE_INSTALL_DATADIR=share/${P}/
+		-DCMAKE_INSTALL_DOCDIR=share/doc/${P}/
```
Comment (SoapGentoo) [inline media-gfx/freecad/freecad-0.16.ebuild]: needs to be `-DCMAKE_INSTALL_DOCDIR=share/doc/${PF}/`, soz for not saying this the last time (https://dev.gentoo.org/~ulm/pms/head/pms.html check the `--docdir` flag)

File: media-gfx/freecad/freecad-0.16.ebuild [hunk 19]
```diff
@@ -0,0 +1,135 @@
...
+	export QT_SELECT=4
+
+#-DOCC_* defined with cMake/FindOpenCasCade.cmake
+#-DCOIN3D_* defined with cMake/FindCoin3D.cmake
+#-DSOQT_ not used
+#-DFREECAD_USE_EXTERNAL_KDL="ON" waiting for Bug 604130 (keyword ~x86)
+	local mycmakeargs=(
+		-DOCC_INCLUDE_DIR="${CASROOT}"/inc
+		-DOCC_LIBRARY_DIR="${CASROOT}"/$(get_libdir)
+		-DCMAKE_INSTALL_PREFIX="${EPREFIX}"/usr/$(get_libdir)/${P}
```
Comment (SoapGentoo) [inline media-gfx/freecad/freecad-0.16.ebuild]: remove `-DCMAKE_INSTALL_PREFIX` completely, no need to specify

**Comment by a17r** [comment]
One more request: Could you try to add USE=webkit and see if you can make it optional? Qt4WebKit is already a liability and should be gotten rid of wherever possible. Buildsystem indicates it could be possible, even if being not exactly clear about the implications for `BUILD_GUI`:

```
    if(BUILD_GUI)
        set(QT_USE_QTOPENGL TRUE)
        set(QT_USE_QTSVG TRUE)
        set(QT_USE_QTUITOOLS TRUE)
        set(QT_USE_QTWEBKIT TRUE)
    endif(BUILD_GUI)

    find_package(Qt4)# REQUIRED

    include(${QT_USE_FILE})

    IF(NOT QT4_FOUND)
        MESSAGE(FATAL_ERROR "=======================\n"
                            "Qt libraries not found.\n"
                            "=======================\n")
    ENDIF(NOT QT4_FOUND)

    IF(NOT QT_QTWEBKIT_FOUND)
        MESSAGE("=====================================================\n"
                "Qt Webkit not found, don't build browser integration.\n"
                "=====================================================\n")
    ENDIF(NOT QT_QTWEBKIT_FOUND)
```

**Pull request closed without merging**